langchain4j/src/main/java/dev/langchain4j/data/document/source/ClassPathSource.java

```diff
@@ -0,0 +1,97 @@
+package dev.langchain4j.data.document.source;
+
+import static dev.langchain4j.data.document.Document.FILE_NAME;
+import static dev.langchain4j.data.document.Document.URL;
+import static dev.langchain4j.internal.ValidationUtils.ensureNotBlank;
+import static dev.langchain4j.internal.ValidationUtils.ensureNotNull;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URL;
+
+import dev.langchain4j.data.document.DocumentSource;
+import dev.langchain4j.data.document.Metadata;
+
+/**
+ * Specialization of a {@link DocumentSource} that knows how to read from the classpath
+ */
+public class ClassPathSource implements DocumentSource {
+    public static final String IS_INSIDE_ARCHIVE = "is_inside_archive";
+    private final URL url;
+    private final ClassLoader classLoader;
+    private final Metadata metadata = new Metadata();
+
+    public ClassPathSource(URL url, ClassLoader classLoader) {
+        this.url = ensureNotNull(url, "url");
+        this.classLoader = classLoader;
```

Comment:
```suggestion
        this.classLoader = ensureNotNull(classLoader, "classLoader");
```

Reply:
BTW, should `ClassLoader` be a part of `ClassPathSource`? It is not used here...